src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -728,7 +728,7 @@ public StringBuilder Append(char[]? value, int startIndex, int charCount)
         /// <param name="value">The string to append.</param>
         public StringBuilder Append(string? value)
         {
-            if (value is not null)
+            if (value != null)
```

Comment:
I think you can revert this check, I was initially blaming Roslyn for extra stloc/ldloc but it turned out the reason was "reverted" args via prefixing that you fixed